cmd/publishing-bot/publisher.go

```diff
@@ -65,27 +64,24 @@ type PublisherMunger struct {
 	config     *github.Config
 	// plog duplicates the logs at glog and a file
 	plog *plog
-	// absolute path to the k8s repos.
-	k8sIOPath string
+	// absolute path to the repos.
+	baseRepoPath string
 	// src branch names that are skipped
 	skippedSourceBranches []string
 }
 
 // New will create a new munger.
 func New(config *github.Config) (*PublisherMunger, error) {
-	// validate config
-	if config.SourceRepo != "kubernetes" {
-		// TODO: externalize the rules below to make it generic
-		return nil, fmt.Errorf("only kubernetes as source repo supported right now")
-	}
-	if config.TargetOrg == "" {
-		return nil, fmt.Errorf("target organization cannot be empty")
-	}
-
 	// create munger
 	p := &PublisherMunger{}
+
+	// set the baseRepoPath
 	gopath := os.Getenv("GOPATH")
-	p.k8sIOPath = filepath.Join(gopath, "src", "k8s.io")
+	// if the SourceRepo is not kubernetes, use github.com as baseRepoPath
+	p.baseRepoPath = filepath.Join(gopath, "src", "github.com")
```

Comment:
can't we use the whole base url everywhere, e.g. k8s.io or github.com/openshift ?

Reply:
for the github issue we need some special logic then of course.

Reply:
just realized this must be `filepath.Join(gopath, "src", "github.com", config.TargetOrg)`